mozci/util/hgmo.py

```diff
@@ -0,0 +1,66 @@
+import requests
+from adr.util.memoize import memoize
+
+
+class HGMO():
+    # urls
+    BASE_URL = "https://hg.mozilla.org/"
+    AUTOMATION_RELEVANCE_TEMPLATE = BASE_URL + "{branch}/json-automationrelevance/{rev}"
+    JSON_TEMPLATE = BASE_URL + "{branch}/rev/{rev}?style=json"
+    JSON_PUSHES_TEMPLATE = BASE_URL + "{branch}/json-pushes?version=2&startID={push_id_start}&endID={push_id_end}"  # noqa
+
+    # instance cache
+    CACHE = {}
+
+    def __init__(self, rev, branch='autoland'):
+        self.rev = rev
+        self.branch = branch
+        if self.branch == 'autoland':
+            self.branch_slug = 'integration/autoland'
+        else:
+            self.branch_slug = self.branch
+
+    @staticmethod
+    def create(rev, branch='autoland'):
+        key = (branch, rev)
+        if key in HGMO.CACHE:
+            return HGMO.CACHE[key]
+        instance = HGMO(rev, branch)
+        HGMO.CACHE[key] = instance
+        return instance
+
+    @memoize
```

Comment:
I wonder if all the memoization is going to make us use too much memory. I guess we can worry about that if it ever happens.

Reply:
I was thinking about this, and we should use `adr`'s caching mechanism (powered by [cachy](https://cachy.readthedocs.io/en/latest/)). That way we can store caches permanently on the file system, redis, memcached, etc.

Though the fs cache still isn't negligible, so there may be an argument to memoize it anyway. Maybe something like redis would be faster.